### PR TITLE
Mnaveau/impedance root different from universe

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

Fix when the root frame is not the universe.

- Add demo where:
    - the root frame is the base
    - the base is fixed
    - the base rotate along roll by 1/10 of a turn
- Add a demo where:
    - the root frame is the base
    - the base is not fixed
    - the robot drop from some heights and bounce up and down.

## How I Tested

[//]: # "Explain how you tested your changes"

I ran the demos on my laptop

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

**Review by jviereck — APPROVED** [review]
Some minor changes, looks good overall.

File: demos/demo_impedance_ctrl_bouncing_cpp.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    # Create a Pybullet simulation environment
+    env = BulletEnvWithGround()
+    robot = env.add_robot(Solo12Robot, useFixedBase=False)
+    pybullet.resetDebugVisualizerCamera(1.3, 100, -35, (0.0, 0.0, 0.0))
+    RobotConfig = Solo12Config
+    pin_robot = robot.pin_robot
+
+    # create the data collector
+    data = DataCollector()
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply demos/demo_impedance_ctrl_bouncing_cpp.py:31]: Well in this demos I wanted to show a little how to use RAI in demos as well.

File: demos/demo_robot_com_ctrl_cpp.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
 )
 from bullet_utils.env import BulletEnvWithGround
 from robot_properties_solo.solo12wrapper import Solo12Robot, Solo12Config
-from robot_properties_bolt.bolt_wrapper import BoltRobot, BoltConfig
+
+# from robot_properties_bolt.bolt_wrapper import BoltRobot, BoltConfig
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply demos/demo_robot_com_ctrl_cpp.py:20]: sorry my bad

File: src/impedance_controller.cpp [hunk 3]
```diff
@@ -116,12 +129,8 @@ void ImpedanceController::run(
                                 end_frame_index_,
                                 pinocchio::LOCAL_WORLD_ALIGNED,
                                 end_jacobian_);
-    pinocchio::getFrameJacobian(pinocchio_model_,
-                                pinocchio_data_,
-                                root_frame_index_,
-                                pinocchio::LOCAL_WORLD_ALIGNED,
-                                root_jacobian_);
-    impedance_jacobian_ = end_jacobian_ - root_jacobian_;
```
[the remark this answers is not in the document]
Comment (MaximilienNaveau) [reply src/impedance_controller.cpp:124]: :+1:

**Pull request merged**